pkg/controller/upgradeconfig/cluster_upgrader.go

```diff
@@ -250,40 +251,61 @@ func CommenceUpgrade(c client.Client, upgradeConfig *upgradev1alpha1.UpgradeConf
 	return true, nil
 }
 
-// TODO This create the maint window for control plane
-func CreateControlPlaneMaintWindow(c client.Client, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
+// Create the maintenance window for control plane
+func CreateControlPlaneMaintWindow(c client.Client, m maintenance.Maintenance, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
+	err := m.Start(time.Now().Add(time.Minute * 90))
+	if err != nil {
+		return false, err
+	}
 
 	return true, nil
 }
 
-// TODO This remove the maint window for control plane
-func RemoveControlPlaneMaintWindow(c client.Client, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
+// Remove the maintenance window for control plane
+func RemoveControlPlaneMaintWindow(c client.Client, m maintenance.Maintenance, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
+	err := m.End()
+	if err != nil {
+		return false, err
+	}
 
 	return true, nil
 }
 
-// TODO This create the maint window for control plane
-func CreateWorkerMaintWindow(c client.Client, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
+// Create the maintenance window for workers
+func CreateWorkerMaintWindow(c client.Client, m maintenance.Maintenance, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
+	configPool := &machineconfigapi.MachineConfigPool{}
+	err := c.Get(context.TODO(), types.NamespacedName{Name: "worker"}, configPool)
+	if err != nil {
+		return false, nil
+	}
+
+	pendingWorkerCount := configPool.Status.MachineCount - configPool.Status.UpdatedMachineCount
+	var maintenanceTimePerNode int32 = 8
```

Comment:
See my earlier comment on line 256. But regardless, for this specific line, would prefer we set it to 
```suggestion
	var maintenanceTimePerNode int32 = 8 * time.Minute
```
here, rather than multiply it by the time.Minute down in the `m.Start()` section - that way it is blatantly obvious when we see it defined what unit of time it represents.